pandas/core/resample.py

```diff
@@ -2495,6 +2501,17 @@ def _get_period_bins(self, ax: PeriodIndex):
 
         return binner, bins, labels
 
+    def _set_grouper(
+        self, obj: NDFrameT, sort: bool = False, *, gpr_index: Index | None = None
+    ) -> tuple[NDFrameT, Index, npt.NDArray[np.intp] | None]:
+        obj, ax, indexer = super()._set_grouper(obj, sort, gpr_index=gpr_index)
+        if isinstance(ax.dtype, ArrowDtype) and ax.dtype.kind in "Mm":
```

Comment:
We don't want to do this in `Grouper`'s implementation, is that right?

Reply:
I only see `_set_grouper` being used in one place outside of resample, and where it is the `ax` return is ignored. So I _think_we could just add this to `Grouper`'s implementation, which reduces a little bit of the complexity, but at the cost of doing extra work that just gets discarded (outside of resample).

Definitely good with this as-is.

Reply:
Ah I see. I'll revisit this once the Grouper attribute deprecation are enforced. I think I can simplify this future once those are enforced.